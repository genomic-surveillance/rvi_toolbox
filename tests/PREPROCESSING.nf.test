nextflow_workflow {

    name "Test Workflow PREPROCESSING"
    script "subworkflows/PREPROCESSING.nf"
    workflow "PREPROCESSING"

    test("Should run without failures") {

        when {
            params {
                run_trimmomatic = true
                run_trf = true
                run_hrr = true
                meta = [
                    id: "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1",
                ]
                reads = [
                    "${projectDir}/tests/test_data/FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_100reads_1.fastq", 
                    "${projectDir}/tests/test_data/FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_100reads_2.fastq"
                ]

                script_src_path="${projectDir}/bin/" 
            }

            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.of(tuple(params.meta, params.reads))
                """
            }

            then {
                assert workflow.success
                assert snapshot(workflow.out).match()
            }
        }
    }
}
