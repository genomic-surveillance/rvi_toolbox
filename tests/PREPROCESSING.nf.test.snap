{
    "read-pairs should be balanced running only trf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "trf_test_1"
                        },
                        "trf_test_1_trf_1.fastq:md5,7fba94172828e1be79d86463cb250c76",
                        "trf_test_1_trf_2.fastq:md5,c66b1a985e36c97f5c0768515d04c193"
                    ]
                ],
                "out_ch": [
                    [
                        {
                            "id": "trf_test_1"
                        },
                        "trf_test_1_trf_1.fastq:md5,7fba94172828e1be79d86463cb250c76",
                        "trf_test_1_trf_2.fastq:md5,c66b1a985e36c97f5c0768515d04c193"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-15T15:52:55.119881"
    },
    "Should run without hrr": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1"
                        },
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_trf_1.fastq:md5,d5be190fb01a4c22a7a6f5e2ce9db162",
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_trf_2.fastq:md5,c68aa5a6e2f3c4434beafae689cb0428"
                    ]
                ],
                "out_ch": [
                    [
                        {
                            "id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1"
                        },
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_trf_1.fastq:md5,d5be190fb01a4c22a7a6f5e2ce9db162",
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_trf_2.fastq:md5,c68aa5a6e2f3c4434beafae689cb0428"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-14T09:29:31.605707"
    },
    "Should run only trimmomatic": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1"
                        },
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_trimmed_1.fastq:md5,d5be190fb01a4c22a7a6f5e2ce9db162",
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_trimmed_2.fastq:md5,c68aa5a6e2f3c4434beafae689cb0428"
                    ]
                ],
                "out_ch": [
                    [
                        {
                            "id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1"
                        },
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_trimmed_1.fastq:md5,d5be190fb01a4c22a7a6f5e2ce9db162",
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_trimmed_2.fastq:md5,c68aa5a6e2f3c4434beafae689cb0428"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-14T09:29:40.096628"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1"
                        },
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_1_clean.fastq:md5,d5be190fb01a4c22a7a6f5e2ce9db162",
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_2_clean.fastq:md5,c68aa5a6e2f3c4434beafae689cb0428"
                    ]
                ],
                "out_ch": [
                    [
                        {
                            "id": "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1"
                        },
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_1_clean.fastq:md5,d5be190fb01a4c22a7a6f5e2ce9db162",
                        "FluH1N1_SARSCoV2_hPIV1_33K_ea_P1_2_clean.fastq:md5,c68aa5a6e2f3c4434beafae689cb0428"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-14T09:29:18.781362"
    },
    "Should run only trf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "trf_test_1"
                        },
                        "trf_test_1_trf_1.fastq:md5,dee10c28fdd092f8118fcc31fa3efb57",
                        "trf_test_1_trf_2.fastq:md5,c66b1a985e36c97f5c0768515d04c193"
                    ]
                ],
                "out_ch": [
                    [
                        {
                            "id": "trf_test_1"
                        },
                        "trf_test_1_trf_1.fastq:md5,dee10c28fdd092f8118fcc31fa3efb57",
                        "trf_test_1_trf_2.fastq:md5,c66b1a985e36c97f5c0768515d04c193"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2025-05-14T13:30:23.984080805"
    }
}